### PR TITLE
Add Pump device type to chef with only mandatory clusters enabled

### PR DESCRIPTION
**Pull request by nekleo**
Add Pump device type (0x0303) with only the mandatory clusters enabled: On/Off, Pump Configuration and Control, Identify.

Testing:
* Created binaries for ESP32-WROVER-KIT: ./chef.py -bc -d rootnode_pump_a811bb33a0 -t esp32 -m -p 0x800a -r
* Commissioned using Android
* Controlled via GHA/Android and Nest Hub Gen 2 and using Pigweed RPC.

### CODE REVIEW

**Review by bzbarsky-apple — APPROVED** [review]

File: examples/chef/devices/rootnode_pump_a811bb33a0.matter [hunk 1]
```diff
@@ -0,0 +1,1555 @@
...
+    ram      attribute featureMap default = 1;
+    ram      attribute clusterRevision default = 0x0001;
+  }
+
+  server cluster Switch {
+    ram      attribute numberOfPositions default = 2;
+    ram      attribute currentPosition;
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
```
Comment (ReneJosefsen) [inline examples/chef/devices/rootnode_pump_a811bb33a0.matter:1469]: Why is the switch cluster enabled on endpoint 0?

File: examples/chef/devices/rootnode_pump_a811bb33a0.matter [hunk 2]
```diff
@@ -0,0 +1,1555 @@
...
+    ram      attribute lastNetworkID;
+    ram      attribute lastConnectErrorValue;
+    ram      attribute featureMap default = 1;
+    ram      attribute clusterRevision default = 0x0001;
+  }
+
+  server cluster DiagnosticLogs {
+    ram      attribute featureMap default = 0;
+    ram      attribute clusterRevision default = 1;
+  }
```
Comment (ReneJosefsen) [inline examples/chef/devices/rootnode_pump_a811bb33a0.matter:1439]: Is the intention to include optional clusters on endpoint 0 or should this also be the most basic config?

**Review by andy31415 — APPROVED** [review]

File: examples/chef/devices/rootnode_pump_a811bb33a0.matter (hunk 2 above)
Comment (nekleo) [reply examples/chef/devices/rootnode_pump_a811bb33a0.matter:1439]: I created a zap file that only has required clusters and attributes per the matter spec for EP0 and EP1, and it worked fine. But I don't have the history as to why EP0 is defined with like this to know if it is OK to commit a zap file set to the most basic config. I'll look into it.

File: examples/chef/devices/rootnode_pump_a811bb33a0.matter (hunk 1 above)
Comment (bzbarsky-apple) [reply examples/chef/devices/rootnode_pump_a811bb33a0.matter:1469]: Switch on EP0 is almost certain a "someone messed it up and then it got copy/pasted" issue...

Comment (nekleo) [reply examples/chef/devices/rootnode_pump_a811bb33a0.matter:1469]: I'm going to change the pump zap file to conform to the spec by eliminating Switch from EP0 and all other clusters that aren't required by the root node device type.

**Review by ReneJosefsen — APPROVED** [review]

**Review by ReneJosefsen — APPROVED** [review]

**Pull request merged**